### ISSUE TITLE
support for nhentai.net as a metadata downloader?

### ISSUE THREAD

**Issue by CuddleBear92** [Pewpews/happypanda#32]
hey, im unsure how you guys feel about http://nhentai.net/ and such but as a metadata downloader it would be pretty great IMO.
its like ex without the strict block with profiles (avid user yet still blocked on ex here (years of usage and even H@H hosting)).
i have seen one program (atleast so far) that supports it and thats freemangadownloader (it would read all tags and such quite fine so it should be an issue for your program either i guess).
the site pretty much has no limits in terms of tags and content (mostly)

**Comment by twiddli** [comment]
Happypanda supports fetching metadata from g.e too (without providing any ex data), if you haven't already noticed that.
Sometime in the near future I'll be adding support for more sites and nhentai will be one of them.

**Comment by CuddleBear92** [comment]
yeah..... i just got some removed content that isnt on g.e anymore and so on that i cant get metadata on...... damn those traps XD
thanks for informing and so on. i really do love your app. going to swap to it from ComicRack real soon.
just parsing of filenames could work a bit better tho.

**Comment by twiddli** [comment]
> just parsing of filenames could work a bit better tho.

Work better, as in, tokenize into more than just title, author and language, or?

**Comment by CuddleBear92** [comment]
nhaa those are fine, i think they Roadmap has it right tho.
- Improve gallery searching
- Allow namespace & tag excluding

i think those would fix the issues i have ATM, most of the time it fails for figure out whats the series name and what is the artist and so on. and the autosearch for metadata also failes because of that, meaning there is alot of manual copy past links going around for me ATM. 

really love your program tho and its great, the integrated tags is a blessing IMO,
quick question: ever thought about writing metadata into the cbz files people have? if you write tags as Genre's it would save locally in the CBZ files. meaning other programs that use ComicRack type metadata would read it without an issue aswell (CR Tags only get saved in the database of CR.... not sure if Happy can save and load it from there tho) 
that way Manga from Happy would work better with programs like ubooquity (server hosted manga/comics) and mobile/tablet apps like Chunky Comic Reader.

sorry if im a bit too much offtopic here..... saw the gitter but that seems scary :P

**Comment by twiddli** [comment]
> and the autosearch for metadata also failes because of that, meaning there is alot of manual copy past links going around for me ATM.

Right now the only thing being used by auto metadata fetcher is a hash string generated from one of a gallery's image. You're right though, in some very rare cases I've been able to find the gallery on ex/g.e by looking for it manually even though it said it didn't find any galleries. I'll write it down for now, that I need to improve auto metadata fetcher too. 

> ever thought about writing metadata into the cbz files people have? if you write tags as Genre's it would save locally in the CBZ files. meaning other programs that use ComicRack type metadata would read it without an issue aswell (CR Tags only get saved in the database of CR.... not sure if Happy can save and load it from there tho) 

No, I haven't really thought about it. I might implement it when I decide to implement automatic gallery file/folder renaming when editing a gallery in Happypanda, but only if it won't be too time consuming, sorry.

I see a lot of people migrating from other programs. I might make an option to import galleries from popular programs such as Calibre and ComicRack, but that task will be low on the priority list.

**Comment by CuddleBear92** [comment]
importing from CR wouldnt really be needed in the first place tho. but if your program could read metadata in cbz files it would help quite a bit in the first place (no need for anything more if someone moves from CR)
you could host all metadata you ever have in the cbz files everything from links, artists, series name, undertitle and tags all within one file. (also characters that is in the issue) 
it would be easier to share aswell as the metadata would just be loaded from the files that have them without a need to look after a g.e link to get the metadata.

**Comment by CuddleBear92** [comment]
one thing i miss in your program ATM is a way to correct metadata downloads (that isnt found right away)

like how the comicvine scrapper for CR works is if it does not find a match right away and the program is unsure about which one is the correct one.
it would show a searchlist after the series name where the user can choose the issue themselfs (with a cover image BTW).
something like this would just make things alot simpler when importing alot of issue at the same time.

image of what i mean:
![](http://i.imgur.com/Ny01Mek.png)

it would auto search normally after the series/issue name and if all fails it would show you this manual input image that would let the user choose the real one in a quick and safe manner.
to be able to manually search with something like this relieves the need of using the site almost entirely IMO. 
and as Happy supports tags aswell you could have that inbedded into the manual search aswell.
it all in all would bring more user-friendlyness to the program as not as much clicking back and forth between the program and the site.

**Comment by twiddli** [comment]
> but if your program could read metadata in cbz files it would help quite a bit in the first place

Never really thought of taking advantage of cbz's like that. I'll do something about it in v0.26.

> it would auto search normally after the series/issue name and if all fails it would show you this manual input image that would let the user choose the real one in a quick and safe manner.
> to be able to manually search with something like this relieves the need of using the site almost entirely IMO.
> and as Happy supports tags aswell you could have that inbedded into the manual search aswell.
> it all in all would bring more user-friendlyness to the program as not as much clicking back and forth between the program and the site.

Sounds very useful, yeah. I'll probably make something similar when improving auto metadata fetcher. Thanks for the suggestion.

**Comment by CuddleBear92** [comment]
thank you for listening :D love you app and want to swap to it for good with my mangas as there is no metadata scrapper at all of those things on CR.
love to support and make more suggestions over time when i think about it (if you are OK with it)
one thing i will say tho, you kinda need an auto update feature in your program tho.

**Comment by twiddli** [comment]
> love to support and make more suggestions over time when i think about it (if you are OK with it)

Please keep them coming..!

> one thing i will say tho, you kinda need an auto update feature in your program tho.

The program will notify you when there is a new update but it won't download and update for you. I've thought about making it do just that but that's currently low priority.

**Comment by CuddleBear92** [comment]
aslong as its telling you its fine. thats the most important part.

will have to use the program more for other input tho. but when i get some i will share :D

**Comment by twiddli** [comment]
> will have to use the program more for other input tho. but when i get some i will share :D

Thank you

**Comment by CuddleBear92** [comment]
a few things that bothers me a bit with the program ATM.
one the popup window we get when the program fails to find the series without help you get a list of choices (which is great (all its missing is an image loaded from the site to compare right away))
but the thing is that the window is so small and you are not able to see the full name of the manga at all, there is no way to resize that window either so we can get the full name of the manga (esp hard when you mark ALOT of your collection to a manual metadata download as it failed to find it the first time)

also in the same window i find search results for other languages than English and/or Japanese. we should have a setting that would block other languages from those searches and the program in general.

this window: 
![](http://i.imgur.com/0brEnMr.png)

**Comment by CuddleBear92** [comment]
you should also have reasons of why say a file is skipped under import. i have no idea why it is from time to time. plz specify in the window that opens up about whats going on no matter what small as it gives the user more input and knowledge about how the program works.

**Comment by CuddleBear92** [comment]
also on the same window as i last talked about, there is no way to cancel out of that window if the search is wrong. you have to pick one even tho i might be really really wrong and not even close to what the manga was which messes up your collection as a hole if it happens alot of times aswell. getting harder and harder to keep track of if that happend alot of times over one set of commands.

**Comment by twiddli** [comment]
> but the thing is that the window is so small and you are not able to see the full name of the manga at all, there is no way to resize that window either so we can get the full name of the manga

Are you talking about the gallery in question or about the list of choices?

> also in the same window i find search results for other languages than English and/or Japanese. we should have a setting that would block other languages from those searches and the program in general.

This is a great idea. I'll add it when improving auto metadata fetcher

> you should also have reasons of why say a file is skipped under import. i have no idea why it is from time to time. plz specify in the window that opens up about whats going on no matter what small as it gives the user more input and knowledge about how the program works.

I remember thinking about doing that when working on that part. I'll do something about it, i guess.

> also on the same window as i last talked about, there is no way to cancel out of that window if the search is wrong. you have to pick one even tho i might be really really wrong and not even close to what the manga was which messes up your collection as a hole if it happens alot of times aswell. getting harder and harder to keep track of if that happend alot of times over one set of commands. 

This is true. I'll add a cancel button promptly. 

Thank you.

**Comment by CuddleBear92** [comment]
@ Are you talking about the gallery in question or about the list of choices?
the list of choices, the window in the picture.

**Comment by twiddli** [comment]
Alright, I've made it so you can see the whole text without scrolling. Thanks for the feedback.

**Comment by CuddleBear92** [comment]
hmmm but more importantly, the original source issue text in the same window. the list you get can be scrolled through and such, but not the one you are trying to match up against. as a user with a large collection trying to regain control its hard to know if that one manga im trying to link metadata too is actually just one or two issue or the hole volume and so on. without being able to read the hole name of the issue im trying to link metadata too its hard to be right 100% of the time for sure.

**Comment by twiddli** [comment]
For now, I've made it so it shows the full title and artist texts when you hover as a tooltip. I might do something later on if there's a need for an improvement.

**Comment by CuddleBear92** [comment]
hey again, i do still love your program over many other things out there.
but i recently run into this program: https://doujindownloader.wordpress.com/
its pretty kick ass IMO on how smooth and fast it handles things without issues (alot of those i have tried do have issues after a while).
one good thing this program does is that it can place a tags.txt file inside the cbz file (or whatever filetype you like) and they are parsed so even your program can read it out right.

one thing as mass downloading/collecting goes is if your program could auto import these text files when the manga gets imported (therefor tagging it correctly from the start) 

i have tried pretty much everything on this list now: https://docs.google.com/spreadsheets/d/1uJzgJBGuY5Jv0xcaPwGD8lAlYueu0Z6Q4HWdy1BDDDg/edit?pli=1#gid=208687356
(how i found you BTW :D)

but i do think the one i linked at the start of this comment works the best for me, im guessing alot of other guys are using this aswell. as a hole, auto importing things like tags.txt files in the manga files does speed up everything as fast as it gets really, it also does not matter if the manga is on g.e or ex either as its just tags in a text file.

im not going to try to sell you the program, but i do think its one of the better ones out there and it works with all sites, even tag support.

the tags.txt is layed out pretty simple inside the cbz files like this: Big Breasts, Uncensored, Nakadashi, Anal, Full Color, Stockings 
for example.

**Comment by CuddleBear92** [comment]
it also supports an Info.txt that works the same way that also contains tags but everything else aswell like Title, Author, Artist, Tags & Description.

like so from a random file i got:
TITLE: [AskRay (Bosshi)] Bosshi's XXX MiX [English] {FAKKU}
AUTHOR/CIRCLE: Askray
ARTIST: Bosshi
TAGS: Twins, Futanari, Elf, Swimsuit, Futanari On Futanari, Dragon, Uncensored, Bestiality, Anal, Bondage, Bikini, Lolicon, Inflation, Incest, Sister, Parasite, Big Penis, Tentacles
DESCRIPTION: 

sorry i didnt have this in the original last comment.... i kinda forgot about it. 
i also think H@H downloads does it this way aswell if i remember correctly.

**Comment by twiddli** [comment]
Certainly interesting. I'll take a look at it and maybe add support before i release the next version (by the end of this week).

**Comment by CuddleBear92** [comment]
:O that would be just perfect. that means anyone that uses that program can download from pretty much any site and have those files written out for them and added to your program. thats great!

**Comment by CuddleBear92** [comment]
since the info.txt file imports didnt make it in the last update.
do you have plans on trying to get a version working on the next one?

**Comment by twiddli** [comment]
Yes, I am

**Comment by CuddleBear92** [comment]
nice to hear that you still have it on your mind :D thanks alot! love you dude! will move over fully once its supported as it would take alot of management work off my load to say the least.

**Comment by CuddleBear92** [comment]
i noticed that you plan a new release soonish here: https://github.com/Pewpews/happypanda/issues/75
do you think the text import will be supported in this release? 
my collection all use this since the downloader imports it that way, and since i download from many sites with many standards then the filename is different from whats on g.e and ex.

really not willing to go over thousand and thousand mangas manually again after already doing it once in ComicRack...

**Comment by twiddli** [comment]
See 97e359a8f83e6b8b71191ef82a29362093a12de0

**Comment by CuddleBear92** [comment]
:O 
Thanks a ton! <3
i finally get to use your program for full and regain control over my collection. thanks a ton!
also spreading the work about JMM as a community manager of sorts for them.
http://japanesemediamanager.github.io/
feel free to join us for a chat or whatever on our discord server linked there aswell. we are always nice.... if you like anime :D

feel free to close the issue i guess as it was the main part of this issue IMO, unless you want it open for all the other things i have been rambling on about :D

**Comment by twiddli** [comment]
The issue will be closed once I have implemented metadata fetching from nhentai.

**Comment by CuddleBear92** [comment]
:) your just making my day, cant wait now. just all to damn exited XD
